solr/core/src/java/org/apache/solr/core/PluginBag.java

```diff
@@ -213,6 +237,24 @@ public PluginHolder<T> put(String name, PluginHolder<T> plugin) {
                 apiBag.register(api, nameSubstitutes);
               }
             }
+
+            // TODO Should we use <requestHandler name="/blah"> to override the path that each
+            //  resource registers under?
+            Collection<Class<? extends JerseyResource>> jerseyApis =
+                apiSupport.getJerseyResources();
+            if (!CollectionUtils.isEmpty(jerseyApis)) {
+              for (Class<? extends JerseyResource> jerseyClazz : jerseyApis) {
+                if (log.isDebugEnabled()) {
+                  log.debug("Registering jersey resource class: {}", jerseyClazz.getName());
+                }
+                jerseyResources.register(jerseyClazz);
+                // See MetricsBeanFactory javadocs for a better understanding of this resource->RH
+                // mapping
+                if (inst instanceof RequestHandlerBase) {
```

Comment:
I really appreciate your deeply thoughtful response!
Any way, this is just a smell; and there was some smell before to this effect, albeit less.

FYI note that the package management system can dynamically add/change handlers without even reloading the core.  It remains to be seen if JAX-RS/Jersey works with that system but I wouldn't invest time in that now.

Reply:
OK, I'll leave this as-is for now, but add exploring my first bullet-point above to my todo list for down the road.  If you think of a better approach, let me know!

>  It remains to be seen if JAX-RS/Jersey works with that system

I spent some time testing this out and was happily surprised to find that JAX-RS worked well for core-level packages/plugins, but didn't work at the container level because of some broad design decisions the package-manager makes at that level. (Specifically, we don't allow container-level "requestHandler" plugins, only 'Api' implementations.  More details [here](https://issues.apache.org/jira/browse/SOLR-16347?focusedCommentId=17586116&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17586116))

Anyway, bit of a mixed story there, but at least some good news.